README.md

```diff
@@ -934,7 +935,9 @@ If the feature is enabled, the feature manager will check the `User`, `Group`, a
 
 Allocation logic is similar to the [Microsoft.Targeting](./README.md#MicrosoftTargeting) feature filter, but there are some parameters that are present in targeting that aren't in allocation, and vice versa. The outcomes of targeting and allocation are not related.
 
-#### Overriding Enabled State with a Variant
+**Note:** To allow allocating feature variants, you need to register `ITargetingContextAccessor`. This could be done by calling `WithTargeting<T>` method.
+
+### Overriding Enabled State with a Variant
 
 You can use variants to override the enabled state of a feature flag. This gives variants an opportunity to extend the evaluation of a feature flag. If a caller is checking whether a flag that has variants is enabled, the feature manager will check if the variant assigned to the current user is set up to override the result. This is done using the optional variant property `StatusOverride`. By default, this property is set to `None`, which means the variant doesn't affect whether the flag is considered enabled or disabled. Setting `StatusOverride` to `Enabled` allows the variant, when chosen, to override a flag to be enabled. Setting `StatusOverride` to `Disabled` provides the opposite functionality, therefore disabling the flag when the variant is chosen. A feature with a `Status` of `Disabled` cannot be overridden.

```

Comment:
Maybe we should add a "why?" here, because there's only a narrow reason they would want to use the overrides. It's if a customer wants to allocate via variants without changing their existing IsEnabled code. Meaning they want to return bool results but have more than 2 "buckets".

Reply:
> Today, feature filters can be used to conditionally enable feature flags. Variants should be able to extend this functionality by also playing a role in the decision to enable a feature flag. A property can be included in a feature flag’s variants giving them the opportunity to override the flag’s enabled state. This is important considering the enabled state of a flag is used in our integration points with ASP.NET Core such as the FeatureGateAttribute

quoted from the [design doc](https://microsoft.sharepoint.com/:w:/r/teams/azconfig/_layouts/15/Doc.aspx?sourcedoc=%7BBA41C52A-B66F-4B9A-A3CA-D5CF6283BE06%7D&file=Variants%20as%20an%20extension%20of%20feature%20flags.docx&action=default&mobileredirect=true&share=IQEqxUG6b7aaS6PK1c9ig74GAQjMj0RepCJ3cjLLW_AeLf0)

This is the original purpose for this design.

> This gives variants an opportunity to extend the evaluation of a feature flag.

I think this sentence is a solid reason but we may come up a good story or a attracting use case of statusOverride.

Reply:
> This is important considering the enabled state of a flag is used in our integration points with ASP.NET Core such as the FeatureGateAttribute

This part is valid, but it's still not a big use case. Essentially, customers can always use a normal flag with filters to achieve whatever boolean targeting they want to achieve. Variants don't offer an advantage to boolean targeting over the standard TargetingFilter. 

One reason someone would use Variants in the bool world is to adding more telemetry "buckets" so they can split their traffic (although this is also not necessary as the split traffic of "true" results should be equal). 

Another reason is if a customer's plan is to call both GetVariant and IsEnabled(like using FeatureGateAttribute) on a single flag, then they might use it, but this is likely overloading what a flag should be in charge of. 

My fear is that users will see this section in the readme and think that they **should** be using it- instead of using it to address a specific problem.